app/javascript/mastodon/features/notifications/request.jsx

```diff
@@ -0,0 +1,143 @@
+import PropTypes from 'prop-types';
+import { useRef, useCallback, useEffect } from 'react';
+
+import { defineMessages, useIntl } from 'react-intl';
+
+import { Helmet } from 'react-helmet';
+
+import { useSelector, useDispatch } from 'react-redux';
+
+import ArchiveIcon from '@/material-icons/400-24px/archive.svg?react';
+import DoneIcon from '@/material-icons/400-24px/done.svg?react';
+import VolumeOffIcon from '@/material-icons/400-24px/volume_off.svg?react';
+import { fetchNotificationRequest, fetchNotificationsForRequest, expandNotificationsForRequest, acceptNotificationRequest, dismissNotificationRequest } from 'mastodon/actions/notifications';
+import Column from 'mastodon/components/column';
+import ColumnHeader from 'mastodon/components/column_header';
+import { IconButton } from 'mastodon/components/icon_button';
+import ScrollableList from 'mastodon/components/scrollable_list';
+
+import NotificationContainer from './containers/notification_container';
+
+const messages = defineMessages({
+  title: { id: 'notification_requests.notifications_from', defaultMessage: 'Notifications from {name}' },
+  accept: { id: 'notification_requests.accept', defaultMessage: 'Accept' },
+  dismiss: { id: 'notification_requests.dismiss', defaultMessage: 'Dismiss' },
+});
+
+const selectChild = (ref, index, alignTop) => {
+  const container = ref.current.node;
+  const element = container.querySelector(`article:nth-of-type(${index + 1}) .focusable`);
+
+  if (element) {
+    if (alignTop && container.scrollTop > element.offsetTop) {
+      element.scrollIntoView(true);
+    } else if (!alignTop && container.scrollTop + container.clientHeight < element.offsetTop + element.offsetHeight) {
+      element.scrollIntoView(false);
+    }
+
+    element.focus();
+  }
+};
+
+export const NotificationRequest = ({ multiColumn, params: { id } }) => {
+  const columnRef = useRef();
+  const intl = useIntl();
+  const dispatch = useDispatch();
+  const notificationRequest = useSelector(state => state.getIn(['notificationRequests', 'current', 'item']));
```

Comment:
There might be a discrepancy between `notificationRequests.current` and the `id` param. This should probably be checked, with something like:
```suggestion
  const notificationRequest = useSelector(state => {
    const item = state.getIn(['notificationRequests', 'current', 'item']);
    return item?.get('id') === id : item : undefined;
  });
```